core/base/src/TROOT.cxx

```diff
@@ -930,8 +933,10 @@ TROOT::~TROOT()
       SafeDelete(fGlobals);
       if (fGlobalFunctions) fGlobalFunctions->Delete();
       SafeDelete(fGlobalFunctions);
-      fEnums->Delete();
-      fClasses->Delete();    SafeDelete(fClasses);     // TClass'es must be deleted last
+      fEnums.load()->Delete();
+
+      // FIXME: Causes segfault in rootcling, debug and uncomment
+      // fClasses->Delete();    SafeDelete(fClasses);     // TClass'es must be deleted last
```

Comment:
FYI.  The  intent was to move the 'good delete' out of the #ifdef and keep the bad one hidden behind it ....